services/ui-src/src/utils/forms/schemas.test.tsx

```diff
@@ -14,15 +14,54 @@ describe("Schemas", () => {
   ];
   const badNumberTestCases = ["abc", "N", "", "123:123", "!@#!@%"];
 
-  const testCase = (testCases: Array<string>, expectedReturn: boolean) => {
+  const goodRatioTestCases = [
+    "1:1",
+    "123:123",
+    "1,234:1.12",
+    "0:1",
+    "1:10,000",
+  ];
+  const badRatioTestCases = [
+    ":",
+    ":1",
+    "1:",
+    "1",
+    "1234",
+    "abc",
+    "N/A",
+    "abc:abc",
+    ":abc",
+    "abc:",
+    "%@#$!ASDF",
+  ];
+
+  const testNumberSchema = (
+    testCases: Array<string>,
+    expectedReturn: boolean
+  ) => {
     for (let testCase of testCases) {
       let test = number().isValidSync(testCase);
       expect(test).toEqual(expectedReturn);
     }
   };
 
-  test("Evalulate Number Schema accepting good and bad input", () => {
-    testCase(goodNumberTestCases, true);
-    testCase(badNumberTestCases, false);
+  const testNumberSchemaWithRatioPassed = (
+    testCases: Array<string>,
+    expectedReturn: boolean
+  ) => {
+    for (let testCase of testCases) {
+      let test = number(true).isValidSync(testCase);
+      expect(test).toEqual(expectedReturn);
+    }
+  };
```

Comment:
This is the same as the `testNumberSchema` method above. Do we need both, and both tests? Could we not just do one?

Reply:
They were slightly different, however I reduced it down to the same function now as I agree that they are too similar.